task/host-sp-comms/src/main.rs

```diff
@@ -927,6 +963,132 @@ impl ServerImpl {
         Ok(())
     }
 
+    fn handle_sprot(
+        &mut self,
+        sequence: u64,
+        command: HostToRotCommand,
+        data: &[u8],
+    ) {
+        // All the checks returning CommsBufTooSmall are most likely overkill
+        // but returning an error is much nicer vs a task crash
+        const SPROT_TRANSFER_SIZE: usize = 512;
+        match command {
+            HostToRotCommand::GetCertificates => {
+                let f = |buf: &mut [u8]| {
+                    let mut cnt = 0;
+                    let cert_chain_len = self
+                        .sprot
+                        .cert_chain_len()
+                        .map_err(AttestDataSprotError::from)?;
+                    let mut buf_idx = 0;
+                    for index in 0..cert_chain_len {
+                        let cert_len = self
+                            .sprot
+                            .cert_len(index)
+                            .map_err(AttestDataSprotError::from)?;
+                        for o in (0..cert_len).step_by(SPROT_TRANSFER_SIZE) {
+                            let idx: usize = buf_idx + o as usize;
+                            let end = usize::min(
+                                SPROT_TRANSFER_SIZE,
+                                (cert_len - o) as usize,
+                            );
+                            if idx >= buf.len() || end >= buf.len() {
```

Comment:
Should this be 
```suggestion
                            if idx + end >= buf.len() {
```

instead (based on the `&mut buf[idx..idx+end]` below)?

Reply:
Yeah good catch